jans-linux-setup/jans_setup/setup_app/utils/base.py

```
@@ -381,19 +382,51 @@ def mylog(logs):
             time.sleep(0.1)
-        except:
-             mylog("Error downloading {}. Download will be re-tried once more".format(url))
-             download_tries += 1
-             time.sleep(1)
+            download_ok = True
+        except URLError:
+            retry_sec = random.randint(1,4)
+            mylog(f"Error downloading {url}. Download will be re-tried once more in {retry_sec} seconds.")
+            download_tries += 1
+            if download_tries < 4:
+                time.sleep(retry_sec)
```

Ruff (0.15.21)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)